Philips.CodeAnalysis.Test/Maintainability/RuntimeFailure/DereferenceNullAnalyzerTest.cs

```diff
@@ -172,6 +172,65 @@ public void Foo()
 			VerifyDiagnostic(testCode, Array.Empty<DiagnosticResult>());
 		}
 
+		[TestMethod]
+		public void DereferenceNullAsExpressionReturnLogicalAndCheckDereferenceTest()
+		{
+			string testCode = @"
+class Foo 
+{{
+  public void Foo()
+  {{
+    string x = String.Empty;
+    object obj = x;
+    string y = obj as string;
+    return (y != null && y.ToString() == @"");
+  }}
+}}
+";
+			VerifyDiagnostic(testCode, Array.Empty<DiagnosticResult>());
+		}
+
+		[TestMethod]
+		public void DereferenceNullAsExpressionIfCheckLogicalOrDereferenceTest()
+		{
+			string testCode = @"
+class Foo 
+{{
+  public void Foo()
+  {{
+    string x = String.Empty;
+    object obj = x;
+    string y = obj as string;
+    if (y == null || y.ToString() == @"")
+    {{
+       string t0 = y.ToString();
```

Comment:
If y == null is true, this would fail, right?

Reply:
Good catch. (It's not related, but agree it's a confusing distraction from the actual test.)